perl/lib/Sanger/CGP/Pindel/Implement.pm

```diff
@@ -333,20 +333,25 @@ sub valid_seqs {
   while(<$FAI_IN>) { push @all_seqs, (split /\t/, $_)[0]; }
   close $FAI_IN;
 
+  my @exclude;
+  my @exclude_patt = [];
   if(exists $options->{'exclude'}) {
-    my @exclude = split /,/, $options->{'exclude'};
-    my @exclude_patt;
-    for my $ex(@exclude) {
-      $ex =~ s/%/.+/;
-      push @exclude_patt, $ex;
-    }
-
-    for my $sq(@all_seqs) {
-      push @good_seqs, $sq unless(first { $sq =~ m/^$_$/ } @exclude_patt);
-    }
+    @exclude = split /,/, $options->{'exclude'};
+  }elsif(exists $options->{'excludef'}){
+    open (my $READ, '<', $options->{'excludef'}) or croak("Error opening exclue contigs file '".$options->{'excludef'}."' for reading: ".$!);
+      while(<$READ>){
+        my $line = $_;
+        chomp($line);
+        push(@exclude, $line);
+      }
+    close($READ);
+  }
+  for my $ex(@exclude) {
+    $ex =~ s/%/.+/;
+    push @exclude_patt, $ex;
```

Comment:
I'm keeping the existing method so as to make everything back compatible. Agree that we need to update at least the docs. This should also work with a file of full contig names rather than 'regexp'.

Reply:
66d7677 Update to in-script docs. Have added a note to the wiki about `%` wildcard.

Reply:
There remains possible weirdness that could confuse somebody, but I think it's unlikely to happen. With sequences `qw( FOO-1 FOO.1 )`, both would be ignored by `-e FOO.1`